packages/fxa-auth-server/lib/routes/oauth/index.js

```diff
@@ -298,18 +301,34 @@ module.exports = (log, config, oauthdb, db, mailer, devices) => {
             grantType: request.payload.grant_type,
             uid,
             ecosystemAnonId,
-            clientId: request.payload.client_id,
+            service: request.payload.client_id,
           });
 
           // This is a bit of a hack, but we emit the `account.signed`
           // event to signal to the flow it has been completed (see flowCompleteSignal).
           // Previously, this event would get emitted on the `/certificate/sign`
           // endpoint however, with the move to sync oauth, this does not happen anymore
           // and we need to "fake" it.
-          await request.emitMetricsEvent('account.signed', {
-            uid: uid,
-            device_id: sessionToken.deviceId,
-          });
+          if (scopeSet.contains(OAUTH_SCOPE_OLD_SYNC)) {
+            // The service for this event gets special-cased to 'sync' when the
+            // client is one of the "old" sync clients so that it matches its
+            // pre-oauth `/certificate/sign` event.
+            // ref: https://github.com/mozilla/fxa/pull/6581#issuecomment-702248031
+            const oldSyncClients = ['firefox-desktop', 'firefox-ios'];
```

Comment:
nit: you may find the code here easier overall to maintain if you put the known client_ids in `oldSyncClients` with comments about what they are, rather than the three-line dynamic lookup by readable name that is used below. I don't feel strongly about it either way, it just look me a few cycles to parse out how this comparison works.

Reply:
I added a configuration value for the old sync client ids and used that. 🤜 🤛